src/test/scala/CacheKeyTestSpec.scala

```diff
@@ -0,0 +1,114 @@
+import org.scalatest._
+
+import scala.sys.process._
+class CacheKeyTest {
+  private var cacheKey1 = Option("cacheKey1")
+  private var cacheKey2 = Option("cacheKey2")
+  private var gitCommitsMade = 0
```

Comment:
This seems fragile. I'd instead grab the commit you start the test on, and reset back to that, rather than trying to count commits.